### DISCUSSION TITLE
BoundingBoxCalculator.cpp compile error

### DISCUSSION THREAD

**Discussion by YarivAtiasIntel** [carla-simulator/carla#4823] [Q&A]
First, adding new sensor to 'CarlaUE4\Plugins\Carla\Source\Carla\Sensor' worked as expected, and I didn't had any compile error.

later, I closed UE4Editor and rebuild Carla module. via visual studio. 
as a result, I started to get compilation errors in file BoundingBoxCalculator.cpp.
I didn't made changes to BoundingBoxCalculator nor to the header it includes, plus I am not including it in my sensor.
so its really weird.

If I undo my changes and launch UE4Editor, then create my sensor with the exact same code, 
everything is working. but If I close the editor and rebuild CarlaUE4, then I get the error again.

Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(19): error C2653: 'carla': is not a class or namespace name
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(19): error C2878: 'rpc': a namespace or class of this name does not exist
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(72): error C2027: use of undefined type 'UBoxComponent'
Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(31): note: see declaration of 'UBoxComponent'
  C:/Yossi/Projects/Carla/carla-me/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(73): error C2027: use of undefined type 'UBoxComponent'
Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(31): note: see declaration of 'UBoxComponent'
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2653: 'crp': is not a class or namespace name
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2065: 'CityObjectLabel': undeclared identifier
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2146: syntax error: missing ';' before identifier 'TagQueried'
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2065: 'TagQueried': undeclared identifier
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2653: 'crp': is not a class or namespace name
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2065: 'CityObjectLabel': undeclared identifier
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(101): error C2146: syntax error: missing ';' before identifier 'InTagQueried'
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(102): error C2065: 'TagQueried': undeclared identifier
Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Util/BoundingBoxCalculator.cpp(102): error C2653: 'crp': is not a class or namespace name

v: 0.9.12
OS- Win 10

**Comment by AleksiNarkilahti** [comment]
Our team also struggled with this same error, configuration and behaviour for few weeks without finding any clear indicator what caused this error. 

We just duplicated the sensor files we had modified, added them to sensor list and it somehow fixed it.
